lib/Alchemy/Phrasea/Controller/Root/AccountController.php

```diff
@@ -299,11 +305,99 @@ public function displayAccount()
 
+        $repo_baskets = $this->getBasketRepository();
+        $baskets = $repo_baskets->findActiveValidationAndBasketByUser($user);
+
+        $apiAccounts = $this->getApiAccountRepository()->findByUser($user);
+
+        $ownerPublisher = $this->getFeedPublisherRepository()->findBy(['user' => $user, 'owner' => true]);
+
+        $ownerFeedbackSend = $this->getValidationSessionRepository()->findby(['initiator' => $user, ]);
+
         return $this->render('account/account.html.twig', [
-            'user'          => $user,
-            'evt_mngr'      => $manager,
-            'notifications' => $manager->list_notifications_available($user),
+            'user'                => $user,
+            'evt_mngr'            => $manager,
+            'notifications'       => $manager->list_notifications_available($user),
+            'baskets'             => $baskets,
+            'api_accounts'        => $apiAccounts,
+            'isOwnerPublisher'    => count($ownerPublisher) ? true : false,
+            'isOwnerFeedbackSend' => count($ownerFeedbackSend) ? true : false,
+        ]);
+    }
+
+    /**
+     * @param Request $request
+     * @return RedirectResponse
+     */
+    public function processDeleteAccount(Request $request)
+    {
+        $user = $this->getAuthenticatedUser();
+
+        if($this->app['conf']->get(['registry', 'actions', 'delete-account-email-notification'])) {
+
+            // send email confirmation
+
+            try {
+                $receiver = Receiver::fromUser($user);
+            } catch (InvalidArgumentException $e) {
+                $this->app->addFlash('error', $this->app->trans('phraseanet::erreur: echec du serveur de mail'));
+
+                return $this->app->redirectPath('account');
+            }
+
+            $token = $this->getTokenManipulator()->createAccountDeleteToken($user, $user->getEmail());
+            $url = $this->app->url('account_confirm_delete', ['token' => $token->getValue()]);
+
+
+            $mail = MailRequestAccountDelete::create($this->app, $receiver);
+            $mail->setButtonUrl($url);
+            $mail->setExpiration($token->getExpiration());
+
+            $this->deliver($mail);
+
+            $this->app->addFlash('info', $this->app->trans('phraseanet::account: A confirmation e-mail has been sent. Please follow the instructions contained to continue account deletion'));
+
+            return $this->app->redirectPath('account');
+
+        } else {
+            $this->doDeleteAccount($user);
+
+            $response = $this->app->redirectPath('homepage', [
+                'redirect' => $request->query->get("redirect")
+            ]);
+
+            $response->headers->clearCookie('persistent');
+            $response->headers->clearCookie('last_act');
+
+            return $response;
+        }
+
+    }
+
+    public function confirmDeleteAccount(Request $request)
+    {
+        if (null !== $tokenValue = $request->query->get('token')) {
```

Comment:
AGAIN : NO YODA. This is NOT an option.
if( ($tokenValue = $request->query->get('token')) !== null)